File: leetcode/src/main/java/leetcode/editor/cn/合并两个有序数组mergeSortedArray.java
/**
 * @Title: 88.合并两个有序数组
 * @TitleSlug: mergeSortedArray
 * @Author: Neo
 * @Date: 2024-04-27 23:51:20
 */
package leetcode.editor.cn;

public class 合并两个有序数组mergeSortedArray {
    public static void main(String[] args) {
        // 测试代码
        // Solution solution = new 合并两个有序数组mergeSortedArray().new Solution();
    }
    
    // leetcode submit region begin(Prohibit modification and deletion)
    class Solution {
        public void merge(int[] nums1, int m, int[] nums2, int n) {
            int cur = m + n - 1, i = m - 1, j = n - 1;
            while (i >= 0 && j >= 0) {
                if (nums1[i] > nums2[j]) {
                    nums1[cur--] = nums1[i--];
                } else {
                    nums1[cur--] = nums2[j--];
                }
            }
            if (j >= 0) {
                System.arraycopy(nums2, 0, nums1, 0, j + 1);
            }
        }
    }
// leetcode submit region end(Prohibit modification and deletion)

}
       

File: leetcode/src/main/python/leetcode/editor/cn/合并两个有序数组mergeSortedArray.py
"""
Title: 88.合并两个有序数组
Title Slug: mergeSortedArray
Author: Neo
Date: 2024-05-03 01:16:11
"""
from typing import List


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        p1, p2, p = m - 1, n - 1, m + n - 1
        while p2 >= 0:
            if p1 >= 0 and nums1[p1] > nums2[p2]:
                nums1[p] = nums1[p1]
                p1 -= 1
            else:
                nums1[p] = nums2[p2]
                p2 -= 1
            p -= 1

# leetcode submit region end(Prohibit modification and deletion)
